src/mechs/ec/crypto.ts

```diff
@@ -139,6 +139,15 @@ export class EcCrypto implements types.IContainer {
 
   public getAlgorithm(p11AlgorithmName: string | number) {
     const mechanisms = this.container.session.slot.getMechanisms();
+    switch (p11AlgorithmName) {  
+      case "ECDSA":
+      case "ECDSA_SHA1":
+      case "ECDSA_SHA256":
+      case "ECDSA_SHA384":
+      case "ECDSA_SHA512":
+        p11AlgorithmName = "ECDSA";
+        break;
+    }
```

Comment:
This `switch..case` forces ECDSA with hash to primitive ECDSA mechanism. What if token doesn't implement `ECDSA` mechanism and uses ECDSA with hash only?

`getAlgorithm` function must filter ECDSA mechanisms and select ECDSA with hash if it's possible or use ECDSA without hash

Reply:
We use the Fortify App and get an Error from this function see => https://github.com/PeculiarVentures/fortify/issues/497

It says "Cannot get PKCS11 EC mechanism by name 'ECDSA_SHA384' ". We're using an Atos Smart Card from D-Trust.

At the moment, the this.container.session.slot.getMechanisms() returns an array where all ECDSA_*** Were Replaced by ECDSA Without Hash. 

I found out algName in Array is only "ECDSA", but mechanism.name === p11AlgorithmName (https://github.com/PeculiarVentures/node-webcrypto-p11/blob/master/src/mechs/ec/crypto.ts#L145) is never true than.

So please review my Issue from https://github.com/PeculiarVentures/fortify/issues/497 to understand my problem there.